tola/auth_pipeline.py

```diff
@@ -1,76 +1,91 @@
+    """
+    Redirect the user to the registration page, if we haven't found
+    a user account yet.
+    """
+    if user:
+        return {'is_new': False}
+    try:
+        user = User.objects.get(first_name=details['first_name'],
+                                last_name=details['last_name'],
+                                email=details['email'])
+        return {
+            'is_new': True,
+            'user': user
         }
-        register_user(data, userprofile)
+    except User.DoesNotExist:
+        current_partial = kwargs.get('current_partial')
+        query_params = 'cus_fname={}&cus_lname={}&cus_email={}&' \
+                       'organization_uuid={}&partial_token={}'.format(
+                        details['first_name'], details['last_name'],
+                        details['email'], details['organization_uuid'],
+                        current_partial.token)
+
+        redirect_url = '/accounts/register/?{}'.format(query_params)
+        return HttpResponseRedirect(redirect_url)
 
 
 def auth_allowed(backend, details, response, *args, **kwargs):
     """
-    The same social-core auth_allowed method but fetching and
-    adding white-listed domains from TolaSites
+    Verifies that the current auth process is valid. Emails,
+    domains whitelists and organization domains are applied (if defined).
     """
-    emails = backend.setting('WHITELISTED_EMAILS', [])
-    domains = backend.setting('WHITELISTED_DOMAINS', [])
+    allowed = False
     static_url = settings.STATIC_URL
-    allowed = True
 
+    # Get whitelisted domains and emails defined in the settings
+    whitelisted_emails = backend.setting('WHITELISTED_EMAILS', [])
+    whitelisted_domains = backend.setting('WHITELISTED_DOMAINS', [])
+
+    # Get the whitelisted domains defined in the TolaSites
     site = get_current_site(None)
     tola_site = TolaSites.objects.get(site=site)
     if tola_site and tola_site.whitelisted_domains:
         tola_domains = ','.join(tola_site.whitelisted_domains.split())
         tola_domains = tola_domains.split(',')
-        domains += tola_domains
+        whitelisted_domains += tola_domains
 
     try:
         email = details['email']
     except KeyError:
         logger.warning('No email was passed in the details.')
         allowed = False
     else:
-        if emails or domains:
-            domain = email.split('@', 1)[1]
-            allowed = email in emails or domain in domains
+        domain = email.split('@', 1)[1]
+        if whitelisted_emails or whitelisted_domains:
+            allowed = (email in whitelisted_emails or domain in
+                       whitelisted_domains)
+
+        # Check if the user email domain matches with one of the org oauth
+        # domains and add the organization uuid in the details
+        if not allowed:
+            org_uuid = Organization.objects.values_list(
+                'organization_uuid', flat=True).filter(
+                oauth_domains__contains=domain)
```

Comment:
What happens if one org has `oauth_domains=hello.com,something.com` and other  `oauth_domains=byehello.com`? It would match both.

Reply:
First of all, they shouldn't have the same domain. I'll change the validation but this case should never happen.